pkgs/development/compilers/cudatoolkit/common.nix

```diff
@@ -151,9 +150,31 @@ stdenv.mkDerivation rec {
     mkdir -p $out/nix-support
     echo "cmakeFlags+=' -DCUDA_TOOLKIT_ROOT_DIR=$out'" >> $out/nix-support/setup-hook
 
-    # Set the host compiler to be used by nvcc for CMake-based projects:
+    # Set the host compiler to be used by nvcc.
+    # FIXME: redist cuda_nvcc copy-pastes this code
+
+    # For CMake-based projects:
     # https://cmake.org/cmake/help/latest/module/FindCUDA.html#input-variables
-    echo "cmakeFlags+=' -DCUDA_HOST_COMPILER=${gcc}/bin'" >> $out/nix-support/setup-hook
+    # https://cmake.org/cmake/help/latest/envvar/CUDAHOSTCXX.html
+    # https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_HOST_COMPILER.html
+
+    # For non-CMake projects:
+    # FIXME: results in "incompatible redefinition" warnings ...but we keep
+    # both this and cmake variables until we come up with a more general
+    # solution
+    # https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#compiler-bindir-directory-ccbin
```

Comment:
can we move these to nix comments to keep them from affecting the drv hash? maybe like

```nix
''
lots of stuff
'' + 
# comment
''
rest of stuff
''
```

Reply:
Yeah, I've been preventing myself from moving all of the other comments out of strings. It's going to take a lot of will to keep that for a separate PR xD